histomicstk/FeatureExtraction.py

```diff
@@ -129,6 +187,325 @@ def FeatureExtraction(Label, I, K=128, Fs=6, Delta=8):
     df['Extent'] = Extent
     df['Solidity'] = Solidity
 
-    # df.insert(0, 'Slide', SlideName)
+    FSDGroup = FSDGroup.reshape(Num, Fs)
+
+    df['FSD1'] = FSDGroup[:, 0]
+    df['FSD2'] = FSDGroup[:, 1]
+    df['FSD3'] = FSDGroup[:, 2]
+    df['FSD4'] = FSDGroup[:, 3]
+    df['FSD5'] = FSDGroup[:, 4]
+    df['FSD6'] = FSDGroup[:, 5]
+
+    df['HematoxlyinMeanIntensity'] = HematoxylinIntensityGroup.MeanIntensity
+    df['HematoxlyinMeanMedianDifferenceIntensity'] \
+        = HematoxylinIntensityGroup.MeanMedianDifferenceIntensity
+    df['HematoxlyinMaxIntensity'] = HematoxylinIntensityGroup.MaxIntensity
+    df['HematoxlyinMinIntensity'] = HematoxylinIntensityGroup.MinIntensity
+    df['HematoxlyinStdIntensity'] = HematoxylinIntensityGroup.StdIntensity
+    df['HematoxlyinEntropy'] = HematoxylinTextureGroup.Entropy
+    df['HematoxlyinEnergy'] = HematoxylinTextureGroup.Energy
+    df['HematoxlyinSkewness'] = HematoxylinTextureGroup.Skewness
+    df['HematoxlyinKurtosis'] = HematoxylinTextureGroup.Kurtosis
+    df['HematoxlyinMeanGradMag'] = HematoxylinGradientGroup.MeanGradMag
+    df['HematoxlyinStdGradMag'] = HematoxylinGradientGroup.StdGradMag
+    df['HematoxlyinEntropyGradMag'] = HematoxylinGradientGroup.EntropyGradMag
+    df['HematoxlyinEnergyGradMag'] = HematoxylinGradientGroup.EnergyGradMag
+    df['HematoxlyinSkewnessGradMag'] \
+        = HematoxylinGradientGroup.SkewnessGradMag
+    df['HematoxlyinKurtosisGradMag'] \
+        = HematoxylinGradientGroup.KurtosisGradMag
+    df['HematoxlyinSumCanny'] = HematoxylinGradientGroup.SumCanny
+    df['HematoxlyinMeanCanny'] = HematoxylinGradientGroup.MeanCanny
```

Comment:
@slee172 Instead of explicity setting each feature with `df[feature_name]=feature_value` statements, you could use a loop as follows:

```
for feature_name in HematoxylinIntensityGroup._fields:
    df['Hematoxylin' + feature_name] = getattr(HematoxylinIntensityGroup, feature_name)
```
This would make the code compact and a bit more cleaner in the sense that in the future you can add a feature within the feature group function and not have to change the root function.

You can do the same or something similar (using a loop) for other feature groups (FSD, Cytoplasm etc) as well.